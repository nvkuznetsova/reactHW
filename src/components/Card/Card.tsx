import React from 'react';

export type CardProps = {
  cardClassName: string,
};

export const Card: React.FC<CardProps> = ({ children, cardClassName }) => (
  <div className={`card mb-3 ${cardClassName}`}>
    {children}
  </div>
);