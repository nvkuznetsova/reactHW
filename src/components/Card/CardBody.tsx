import React from 'react';

export const CardBody: React.FC = ({ children }) => (
  <div className="card-body">
    {children}
  </div>
);