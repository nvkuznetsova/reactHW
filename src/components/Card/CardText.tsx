import React from 'react';

export type CardTextProps = {
  cardText: string,
};

export const CardText: React.FC<CardTextProps> = ({ cardText }) => (
  <p className="card-text">{cardText}</p>
);
