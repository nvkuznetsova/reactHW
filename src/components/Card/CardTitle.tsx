import React from 'react';

import { Badge } from 'src/components/Badge/Badge'

type CardTitleProps = {
  name: string,
  badgeText: string,
  badgeClassName: string,
};

export const CardTitle: React.FC<CardTitleProps> = ({ name, badgeText, badgeClassName }) => (
  <h4 className="card-title">
    {name}
    <Badge 
      badgeText={badgeText}
      badgeClassName={badgeClassName}
    />
  </h4>
);