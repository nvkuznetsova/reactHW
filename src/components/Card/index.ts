export { Card } from './Card';
export { CardBody } from './CardBody';
export { CardText } from './CardText';
export { CardTitle } from './CardTitle';