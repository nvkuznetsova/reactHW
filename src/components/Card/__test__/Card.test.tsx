import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { Card, CardProps } from '../Card';

describe('Card', () => {
  const getProps = (): CardProps => ({
    cardClassName: 'card-class', 
  });
  const setup = (props = {}): ShallowWrapper => shallow(<Card {...getProps()} {...props} />);

  it('should match snapshot', () => {
    const children = <div>Card Body</div>
    const component = setup({ children });
    expect(component).toMatchSnapshot()
  });
});