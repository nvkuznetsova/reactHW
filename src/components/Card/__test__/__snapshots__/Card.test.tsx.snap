// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div
  className="card mb-3 card-class"
>
  <div>
    Card Body
  </div>
</div>
`;
